test/asserts/big-number-assert_test.js

```diff
@@ -20,31 +20,71 @@ const Assert = BaseAssert.extend({
  */
 
 describe('BigNumberAssert', () => {
-  it('should throw an error if the input value is not a big number', () => {
-    const choices = [[], {}, ''];
+  [undefined, { validateSignificantDigits: true }, { validateSignificantDigits: false }].forEach(option => {
+    describe(`with option '${option ? `{ validateSignificantDigits: ${option.validateSignificantDigits} }` : undefined }'`, () => {
+      it('should throw an error if the input value is not a big number', () => {
+        const choices = [[], {}, ''];
 
-    choices.forEach(choice => {
-      try {
-        new Assert().BigNumber().validate(choice);
+        choices.forEach(choice => {
+          try {
+            new Assert().BigNumber(option).validate(choice);
 
-        should.fail();
-      } catch (e) {
-        e.should.be.instanceOf(Violation);
-      }
-    });
-  });
+            should.fail();
+          } catch (e) {
+            e.should.be.instanceOf(Violation);
+          }
+        });
+      });
 
-  it('should expose `assert` equal to `BigNumber`', () => {
-    try {
-      new Assert().BigNumber().validate();
+      it('should expose `assert` equal to `BigNumber`', () => {
+        try {
+          new Assert().BigNumber(option).validate();
 
-      should.fail();
-    } catch (e) {
-      e.show().assert.should.equal('BigNumber');
-    }
-  });
+          should.fail();
+        } catch (e) {
+          e.show().assert.should.equal('BigNumber');
+        }
+      });
+
+      it('should accept a big number', () => {
+        const choices = [
+          1.01,
+          1.0111111111,
+          '1.0111111111'
+        ];
 
-  it('should accept a big number', () => {
-    new Assert().BigNumber().validate(1.01);
+        choices.forEach(choice => {
+          new Assert().BigNumber(option).validate(choice);
+        });
+      });
+
+      if (!option || option.validateSignificantDigits === true) {
+        it('should throw an error if a number has more than 15 significant digits and `validateSignificantDigits` is true or missing', () => {
```

Comment:
"and `validateSignificantDigits` is true or missing" is now redundant, since this is scoped under `describe('with option '${option}')`.

Same for "if `validateSignificantDigits` is false" below.

Reply:
👍